persistcache/persistcache.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package persistcache
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+type persistCache struct {
+	cache map[string]string
+	root  string
+}
+
+const FILE_MASK = 0755
+
+// Load values from cache or creates path if there's none
+func Load(path string) (*persistCache, error) {
+	pc := &persistCache{}
+	pc.root = path
+	pc.cache = make(map[string]string)
+
+	if _, err := os.Stat(pc.root); os.IsNotExist(err) {
+		os.MkdirAll(pc.root, FILE_MASK)
+		return pc, nil
+	}
+
+	err := filepath.WalkDir(pc.root, func(path string, di fs.DirEntry, err error) error {
+		// We skip all directories
+		if di.IsDir() {
+			return nil
+		}
+
+		val, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		pc.cache[di.Name()] = string(val)
+
+		return nil
+	})
+	return pc, err
+}
+
+// Get value from cache
+func (pc *persistCache) Get(key string) (string, bool) {
+	val, ok := pc.cache[key]
```

Comment:
This is handled, since I use mutexes, we can actually use RWMutex, since that's the case

Reply:
Where do you use mutexes? All accesses (set, delete and references) to the map need to be single-threaded to avoid the go runtime panicing due to concurrent access.